src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestConfigureCallGen.generated.txt

```diff
@@ -1,91 +1,124 @@
+            }
         }
 
-        if (configuration["MyInt"] is string stringValue2)
+        public static void BindCore(IConfiguration configuration, ref Program.MyClass obj)
         {
-            obj.MyInt = int.Parse(stringValue2);
-        }
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
 
-        System.Collections.Generic.List<int> temp3 = obj.MyList;
-        temp3 ??= new System.Collections.Generic.List<int>();
-        BindCore(configuration.GetSection("MyList"), ref temp3);
-        obj.MyList = temp3;
+            if (configuration["MyString"] is string stringValue4)
+            {
+                obj.MyString = stringValue4;
+            }
 
-        System.Collections.Generic.Dictionary<string, string> temp4 = obj.MyDictionary;
-        temp4 ??= new System.Collections.Generic.Dictionary<string, string>();
-        BindCore(configuration.GetSection("MyDictionary"), ref temp4);
-        obj.MyDictionary = temp4;
+            if (configuration["MyInt"] is string stringValue5)
+            {
+                obj.MyInt = int.Parse(stringValue5);
+            }
 
-    }
+            List<Int32> temp6 = obj.MyList;
+            temp6 ??= new List<Int32>();
+            BindCore(configuration.GetSection("MyList"), ref temp6);
+            obj.MyList = temp6;
 
-    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.List<int> obj)
-    {
-        if (obj is null)
-        {
-            throw new global::System.ArgumentNullException(nameof(obj));
+            Dictionary<String, String> temp7 = obj.MyDictionary;
+            temp7 ??= new Dictionary<String, String>();
+            BindCore(configuration.GetSection("MyDictionary"), ref temp7);
+            obj.MyDictionary = temp7;
         }
 
-        int element;
-        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
+        public static bool HasValue(IConfiguration configuration)
         {
-            if (section.Value is string stringValue5)
+            if ((configuration as IConfigurationSection)?.Value is not null)
             {
-                element = int.Parse(stringValue5);
-                obj.Add(element);
+                return true;
             }
-        }
-    }
-
-    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.Dictionary<string, string> obj)
-    {
-        if (obj is null)
-        {
-            throw new global::System.ArgumentNullException(nameof(obj));
-        }
 
-        string key;
-        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
-        {
-            if (section.Key is string stringValue6)
+            foreach (IConfigurationSection section in configuration.GetChildren())
```

Comment:
(something to think about outside of this PR) I almost wonder if we should make a `bool HasChildren` API... That way we don't need to allocate here just to see if the config has children.

Reply:
I think it's a good idea. I can create an issue.

Reply:
https://github.com/dotnet/runtime/issues/83692